shell/platform/common/cpp/client_wrapper/include/flutter/event_stream_handler.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+
+#include "event_sink.h"
+
+namespace flutter {
+
+template <typename T>
+struct StreamHandlerError {
+  const std::string& error_code;
+  const std::string& error_message;
+  const T* error_details;
+
+  StreamHandlerError(const std::string& error_code,
+                     const std::string& error_message,
+                     const T* error_details)
+      : error_code(error_code),
+        error_message(error_message),
+        error_details(error_details) {}
+};
+
+// Handler of stream setup and tear-down requests.
+// Implementations must be prepared to accept sequences of alternating calls to
+// onListen() and onCancel(). Implementations should ideally consume no
+// resources when the last such call is not onListen(). In typical situations,
+// this means that the implementation should register itself with
+// platform-specific event sources onListen() and deregister again onCancel().
+template <typename T>
+class StreamHandler {
+ public:
+  StreamHandler() = default;
+  virtual ~StreamHandler() = default;
+
+  // Prevent copying.
+  StreamHandler(StreamHandler const&) = delete;
+  StreamHandler& operator=(StreamHandler const&) = delete;
+
+  // Handles a request to set up an event stream. Returns error if representing
+  // an unsuccessful outcome of invoking the method, possibly nullptr.
```

Comment:
I'm finding this sentence a bit hard to parse. How about just "Returns nullptr on success, or an error on failure."

Reply:
Thank you! Modified.